test/01_contraction/contraction_resource.cpp

```diff
@@ -66,6 +66,18 @@ namespace hiptensor
     {
     }
 
+    ContractionResource::~ContractionResource()
+    {
+        Base::freeDevice(mDeviceA);
```

Comment:
The host and device pointers are unique_ptr objects which are (should be) self-managed resources. Is this necessary?